standard/basic-concepts.md

```diff
@@ -82,12 +82,14 @@ There are several different types of declaration spaces, as described in the fol
+- Additional local variable declaration spaces may occur within member declarations, anonymous functions and local functions. Names are introduced into these declaration spaces through *pattern*s, *declaration_expression*s,  *declaration_statement*s and *exception_specifier*s. Local variable declaration spaces may be nested, but it is an error for a local variable declaration space and a nested local variable declaration space to contain elements with the same name. Thus, within a nested declaration space it is not possible to declare a local variable, local function or constant with the same name as a parameter, type parameter, local variable, local function or constant in an enclosing declaration space. It is possible for two declaration spaces to contain elements with the same name as long as neither declaration space contains the other. Local declaration spaces are created by the following constructs:
   - Each *variable_initializer* in a field and property declaration introduces its own local variable declaration space, that is not nested within any other local variable declaration space.
   - The body of a function member, anonymous function, or local function, if any, creates a local variable declaration space that is considered to be nested within the function’s local variable declaration space.
   - Each *constructor_initializer* creates a local variable declaration space nested within the instance constructor declaration. The local variable declaration space for the constructor body is in turn nested within this local variable declaration space.
-  - Each *block*, *switch_block*, *iteration_statement* and *using_statement* creates a nested local variable declaration space.
+  - Each *block*, *switch_block*, *specific_catch_clause*, *iteration_statement* and *using_statement* creates a nested local variable declaration space.
   - Each *embedded_statement* that is not directly part of a *statement_list* creates a nested local variable declaration space.
+  - Each *switch_section* creates a nested local variable declaration space. However, variables declared directly within the *statement_list* of the *switch_section* (but not within a nested local variable declaration space inside the *statement_list*) are added directly to the local variable declaration space of the enclosing *switch_block*, instead of that of the *switch_section*.
```

Comment:
I know you called this “subtle”, I might use another term… That said the description of this “feature” looks correct here, however I think an example or note would be particularly worthwhile here

Reply:
+1 for having a note, although I don't think that should block C# 7.

I note that in "The C# Programming Language 4th edition (Special Annotated Edition)" there was a suggestion that it's a pity the original C# designers hadn't redesigned switch/case more thoroughly...

Reply:
I rewrote this particular bullet many times and did many tests to assure myself that it was correct. But straightforward it is not! 😄

I agree in principle on the note, but the problem is that I wouldn't know what to write in it!

Reply:
Agree this doesn't need to block 7.3, but maybe an issue for "pre 8" to add an example?

Reply:
@Nigel-Ecma: I'll add that now.